batch_job/src/work_queue_factory.c

```diff
@@ -344,17 +346,30 @@ static int submit_worker( struct batch_queue *queue )
 		);
 	}
 	else {
-		cmd = string_format(
-		"./%s %s %d -t %d -C '%s' -d all -o worker.log %s %s %s",
-		worker,
-		master_host,
-		master_port,
-		worker_timeout,
-		catalog_host,
-		password_file ? "-P pwfile" : "",
-		resource_args ? resource_args : "",
-		extra_worker_args ? extra_worker_args : ""
-		);
+		if(k8s_worker_image) {
```

Comment:
I see what you are trying to here, but structuring the code this way is going to lead to confusion down the road and difficulty in merging, since these if-branches are practically identical.

How about just this instead:

```
if( kubernetes mode ) {
   worker = "work_queue_worker"
}
```

And then leave the existing string_format as-is.

Reply:
Yes. It is more reasonable to change the value of  `char *worker` .